TTS/utils/synthesizer.py

```diff
@@ -356,7 +358,10 @@ def tts(
         if speaker_wav is not None and self.tts_model.speaker_manager is not None:
             speaker_embedding = self.tts_model.speaker_manager.compute_embedding_from_clip(speaker_wav)
 
+        vocoder_device = "cpu"
         use_gl = self.vocoder_model is None
+        if not use_gl:
+            vocoder_device = next(self.vocoder_model.parameters()).device
```

Comment:
In some obscure use cases, the user could have placed the feature frontend and the vocoder on different devices.

```python
>>> tts.synthesizer.tts_model = tts.synthesizer.tts_model.to("cuda:0")
>>> tts.synthesizer.vocoder_model = tts.synthesizer.vocoder_model.to("cuda:1")
```

We check the device of the vocoder, if it exists.